v0.11 documentation (#33)
iter8-tools/docs: docs/getting-started/concepts.md

---

# Iter8

> Iter8 is the Kubernetes testing platform built for DevSecOps, MLOps, SRE and data science teams. Iter8 makes it easy to ensure that new versions of apps and ML models perform well and maximize business value.

Reviewer: > Iter8 is the Kubernetes testing platform built for DevSecOps, MLOps, SRE and data science teams.

Do we want to say "**the** Kubernetes testing platform" or "**a** Kubernetes testing platform"?

Author: I prefer **the** ; **the one and only** is really I want to see but that is a bit too cheesy :)

## Iter8 Experiment
<p align='center'>
  <img alt-text="load-test-http" src="../../images/iter8-intro-dark.png" width="70%" />
</p>

> Iter8 experiments makes it simple to collect performance and business metrics for apps and ML models, assess and compare one or more app/ML model versions, validate [service-level objectives (SLOs)](#service-level-objectives), promote the winning version, and maximize business value during each release.

Reviewer: ```suggestion
Iter8 experiments make it simple to collect performance and business metrics for apps and ML models, assess and compare one or more app/ML model versions, validate [service-level objectives (SLOs)](#service-level-objectives), promote the winning version, and maximize business value during each release.
```

Author: Fixed.

Specifying an Iter8 experiment involves specifying the list of [tasks](#tasks) executed during the experiment, their [parameters](../user-guide/topics/parameters.md), and the experiment [runner](#loops-and-runner).

### Tasks

> An experiment is an ordered collection of tasks that are executed sequentially. Iter8 provides pre-defined tasks for various functions such as generating load and collecting Iter8's built-in metrics for HTTP and gRPC services, collecting custom metrics for one or more versions of an app from databases, assessing [SLOs](#service-level-objectives), and checking if the application is ready.

Reviewer: ```suggestion
An experiment is an ordered collection of tasks that are executed sequentially. Iter8 provides pre-defined tasks for various functions such as generating load and collecting Iter8's built-in metrics for HTTP and gRPC services, collecting custom metrics from databases, assessing [SLOs](#service-level-objectives), and checking if the application is ready.
```

This is a pretty long sentence so I was wondering if we could shorten it a bit.

Author: Fixed.

### Loops

> A single loop of an experiment involves each task in the experiment executing once. Iter8 experiments can be set up to loop once and finish (single-loop), or loop repeatedly over time (multi-loop).

Reviewer: ```suggestion
A basic experiment involves each task in the experiment executing once but Iter8 experiments can also loop, meaning they will executing tasks more than once. You can configure experiments to loop once and finish (single-loop), or loop repeatedly over time (multi-loop).
```

Author: Fixed.

### Execution Environments
Experiments can run inside Kubernetes clusters, in the local environment, and within CI/CD/GitOps workflows including GitHub Actions workflows.

#### Runners

> A single-loop experiment that is run inside a Kubernetes cluster uses the Kubernetes [job](https://kubernetes.io/docs/concepts/workloads/controllers/job/) workload as its runner. A multi-loop experiment that is run inside a Kubernetes cluster uses the Kubernetes [cronjob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) as its runner.

Reviewer: Should this section be somehow combined with the loops section?

---

# Iter8
Iter8 is the Kubernetes release optimizer built for DevOps, MLOps, SRE and data science teams. Iter8 makes it easy to ensure that Kubernetes apps and ML models perform well and maximize business value.

## Iter8 experiment
<p align='center'>
  <img alt-text="load-test-http" src="../../images/iter8-intro-dark.png" width="70%" />
</p>

Iter8 experiments make it simple to collect performance and business metrics for apps and ML models, assess and compare one or more app/ML model versions, validate [service-level objectives (SLOs)](#service-level-objectives), promote the winning version, and maximize business value during each release.

### Tasks
An experiment is a set of tasks that are executed in a specific sequence. Iter8 provides pre-defined tasks for various functions such as the following:

1. Generating load and collecting Iter8's built-in metrics for HTTP services.
2. Generating load and collecting Iter8's built-in metrics for gRPC services.

> 3. Fetching metrics for one or more versions of an app from databases. 

Reviewer 2: I wonder if we can/should be more explicit: from metrics databases.  _or_ from REST accessible metrics stores.

Author: Fixed.

4. Assessing [SLOs](#service-level-objectives) for app versions.
5. Checking if app is ready.

### Loops
A single loop of an experiment involves the entire sequence of [tasks](#tasks) in the experiment executing once. You can configure experiments to loop once and finish (**single-loop**), or loop repeatedly over time (**multi-loop**).

### Execution environments
Experiments can run inside Kubernetes clusters, in the local environment, and within CI/CD/GitOps workflows including GitHub Actions workflows. Experiments that are executed inside Kubernetes clusters are referred to as **Kubernetes experiments**. All other experiments are referred to as **local experiments**.

> ### Runner

Reviewer 2: Perhaps we should title this "Kubernetes Runner"; currently this is kubernetes centric.

---

# Iter8

> Iter8 is the Kubernetes release optimizer built for DevOps, MLOps, SRE and data science teams. Iter8 makes it easy to ensure that Kubernetes apps and ML models perform well and maximize business value.

Reviewer: ... whereas this says "a Kubernetes release optimizer".

Reviewer: ... whereas this says "DevOps".
